docs/content/guide/component.ngdoc

```diff
@@ -233,9 +233,13 @@ it upwards to the heroList component, which updates the original data.
 </file>
 
 <file name="heroDetail.js">
-  function HeroDetailController($scope, $element, $attrs) {
+  function HeroDetailController($scope) {
     var ctrl = this;
 
+    ctrl.hero = ctrl.hero || {};
+    ctrl.hero.name = ctrl.hero.name || undefined;
+    ctrl.hero.location = ctrl.hero.location || 'unknown';
```

Comment:
Those changes were tested in the unit tests. The tests were failing because the actual functionality was missing. I chose to update the component instead of removing the failing test. If needed, I can remove the test and the functionality.